### ISSUE TITLE
Python 3.5?

### ISSUE THREAD

**Issue by jnmclarty** [dsuch/pymqi#40]
Hi all,

I'm curious if anybody has attempted building this for Python 3.5?  Anybody looked into how hard upgrading the compatability for 3.5 would be?  If the maintainers are working on this, or if they would welcome a pull request on the topic?  What's the appetite?

I haven't even tried, but going to likely be attempting in the days ahead.

**Comment by dsuch** [comment]
Hello @jnmclarty,

yes, there is definitely interest in it!

If you can add it, really the only thing would be to keep in mind Python 2.7 compatibility.

I believe we can drop support for 2.6 and earlier if need be, but 2.7 is still a must.

Kind regards.

**Comment by jnmclarty** [comment]
If I do the overhaul to make a 2.7+3.5 compatible package, are you in a position to be able to support python 3.5 for 64-bit linux going forward? Myself, I'm not an MQ expert.  Are you versed enough in 3.5, to maintain it into the future?  Or, prepared to learn/modify your test stack accordingly?  Part of my PR would be adding details to document preparing a test suite for an automated 2.7 + 3.5 combined suite using tox.  Do you know tox?  Or would you be prepared to learn it?

**Comment by dsuch** [comment]
Hi @jnmclarty - yes, I have been maintaining and developing PyMQI for the last 10 years. I am an MQ expert. I am versed enough in Python to maintain PyMQI for the foreseeable future. I use tox in various places, no worries. The only reason that we don't have Python 3 support is that I spend most of my time at work on [Zato](https://zato.io/docs/).

It's OK to add dependencies to the test suite that best fulfill a given role. As for the core - I would prefer not to have any dependencies because people use PyMQI in environments that don't have network connectivity or work with admins who for various reasons require packages to be vetted manually and such process can take months or years.

Regarding the test suite - feel free to author it in a way that will require a running MQ installation. The wire protocol is proprietary, can change at any time (however unlikely it is) so it is fine to expect a queue manager run during the tests as long as it does not require a lot of preparations and if it does, it would be best to have MQSC scripts to set up test objects upfront, for instance.

Simply put, let's talk about it and design it together in GitHub or if you'd like to contact me directly feel free to reach me at dsuch@zato.io.

**Comment by dsuch** [comment]
Hello @jnmclarty - does your offer still stand or not really due to other pressing needs?

**Comment by dsuch** [comment]
Closing due to inactivity and in favour of #13. Thanks @jnmclarty!
